Replace use of `first_layer` in init with `encoder_layer` argument to init

mikekgfb opened the pull request:
Summary:
Replace use of `first_layer` in init with `encoder_layer` argument to init
(better eng)

Test Plan: sandcastle, github CI

Differential Revision: D46940537

mikaylagawarecki commented:
@pytorchbot merge -r

mikaylagawarecki commented:
@mikekgfb could you rebase and re-export this to land please, seems like the bot is erroneously detecting internal changes

Commit:
Replace use of first_layer` in init with the passed in encoder_layer (#104058)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/104058

Replace use of first_layer` in init with the passed in encoder_layer
(better eng)

Test Plan: sandcastle, github CI

Reviewed By: mikaylagawarecki

Differential Revision: D46940537

fbshipit-source-id: 778fe1a471893c282f6a86c0409e65f0bd9a0fcc

mikekgfb commented:
@pytorchbot merge
